src/DotNetWorker.ApplicationInsights/Initializers/TokenCredentialOptions.cs

```diff
@@ -42,64 +42,64 @@ public static TokenCredentialOptions ParseAuthenticationString(string applicatio
 
             var tokenCredentialOptions = new TokenCredentialOptions();
             bool isValidConfiguration = false;
+
             foreach ((int, int) split in Tokenize(applicationInsightsAuthenticationString))
             {
                 (int start, int length) = split;
 
-                // Trim whitespace from start
-                while (length > 0 && char.IsWhiteSpace(applicationInsightsAuthenticationString[start]))
-                {
-                    start++;
-                    length--;
-                }
+                var authenticationStringToken = applicationInsightsAuthenticationString
+                    .AsSpan(start, length)
+                    .Trim();
```

Comment:
Used to trim only start, but now we trim both ends? Probably fine

Reply:
This removes the need to trim again in line 80 (original code). Here, we just trim in one shot (pretty cheap operation with spans) and don't have to worry about doing that again.